code/modules/mob/living/basic/vermin/frog.dm

```diff
@@ -75,6 +75,23 @@
 		if(L.mob_size > MOB_SIZE_TINY)
 			playsound(src, stepped_sound, 50, TRUE)
 
+/mob/living/basic/frog/frog_suicide
+	name = "suicide frog"
+	desc = "Driven by sheer will."
+	icon_state = "frog_trash"
+	icon_living = "frog_trash"
+	icon_dead = "frog_trash_dead"
+	maxHealth = 5
+	health = 5
+	ai_controller = /datum/ai_controller/basic_controller/frog/suicide_frog
+	///how long do we exist for
+	var/existence_period = 15 SECONDS
+
+/mob/living/basic/frog/frog_suicide/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/explode_on_attack, mob_type_dont_bomb = typecacheof(list(/mob/living/basic/frog, /mob/living/basic/leaper)))
+	AddElement(/datum/element/temporary_atom, life_time = existence_period)
```

Comment:
This _works_ but I think the "flavour" of how `temporary_atom` works is a bit sad for this.
By this I mean they "fade out" once they reach the end of their lifetime and vanish, but ideally these guys should die or gib.
Meanwhile these should probably like, gib instead. Or at least die (but that will pile up a lot of corpses).

Maybe make an element or component for "die after x seconds" and apply it to both this and legion/hivelord spawn?